src/views/connectionUI.ts

```diff
@@ -490,9 +537,20 @@ export class ConnectionUI {
         return ConnectionProfile.createProfile(this._prompter);
     }
 
-    public promptForRetryCreateProfile(profile: IConnectionProfile): PromiseLike<IConnectionProfile> {
+    private async promptToRetryAndSaveProfile(profile: IConnectionProfile, isFirewallError: boolean = false): Promise<IConnectionProfile> {
+        return this.promptForRetryCreateProfile(profile, isFirewallError).then(updatedProfile => {
```

Comment:
This is already async - you can get rid of the .then and just await on promptForRetryCreateProfile